x-pack/plugins/observability_solution/apm/public/hooks/use_eco_tour.tsx

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useLocalStorage } from './use_local_storage';
+
+type TourState = 'shownBefore' | 'isActive';
+
+const INITIAL_STATE: Record<TourState, boolean> = {
+  shownBefore: false,
+  isActive: false,
+};
+
+export function useServiceEcoTour() {
```

Comment:
Like we discussed offline, I think you should split the logic and handle the Modal and Tour visibility separately.

Reply:
updated here https://github.com/elastic/kibana/pull/188918/commits/f09031be87b61ff204f404ea72510e0d6847ccd8#diff-9ab92648a2b049fb46040fee9fb3debfe0bdef33aedc7a063f7c95c3c6e3d696R13